src/Aspire.Hosting.Azure/AzureBicepResource.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Azure;
+
+public class AzureBicepResource(string name, string templateFile) : Resource(name), IAzureResource
+{
+    public string TemplateFile { get; } = templateFile;
```

Comment:
This will need to be more ellaborate. There are some deployment scenarios where you need to be able to use modules/sub deployments to solve a problem. A common example is when you are dealing wtih role assignments and/or tags. If you want to _add a tag_ without destructively removing ones you need a sub deployment. To work around some limitations with the role assignment resource you'll sometimes need a sub module.

I think the way to do this is that we have a method that we pass in a BicepEmissionContext (or something) where you can attach files to it (with a primary entrypoint for a specific resource). It would be the job of the manifest writer (or provisioner in this case) to write those files to disk.

Reply:
An example would be helpful.